statsmodels/stats/anova.py

```diff
@@ -362,6 +367,198 @@ def anova_lm(*args, **kwargs):
     return table
 
 
+class AnovaRM(object):
+    """
+    Repeated measures ANOVA using least square regression
+
+    The full model regression residual sum of squared is
+    used to compared with reduced model for calculating the
+    within subject effect sum of squared [1]
+
+    Between subject effect and Greenhouse-Geisser correction is not yet
+    supported.
+
+    Parameters
+    ----------
+    data : DataFrame
+    dv : string
+        The dependent variable in `data`
+    subject : string
+        Specify the subject id
+    within : a list of string(s)
+        The within-subject factors
+    between : a list of string(s)
+        The between-subject factors
+    subject : string
+        Specify the subject id
+
+    Returns
+    -------
+    ANOVAResults
+
+    References
+    ----------
+    .. [1] Rutherford, Andrew. ANOVA and ANCOVA: a GLM approach. John Wiley & Sons, 2011.
+
+    """
+
+    def __init__(self, data, dv, subject, within=None, between=None):
+        self.data = data
+        self.dv = dv
+        self.within = within
+        if 'C' in within:
+            raise ValueError("Factor name cannot be 'C'! This is in conflict "
+                             "with patsy's contrast function name.")
+        self.between = between
+        if between is not None:
+            raise NotImplementedError('Between subject effect not '
+                                      'yet supported!')
+        self.subject = subject
+        self._check_data_balanced()
+
+    def _check_data_balanced(self):
+        factor_levels = 1
+        for wi in self.within:
+            factor_levels *= len(self.data[wi].unique())
+
+        cell_count = {}
+        for index in range(self.data.shape[0]):
+            key = []
+            for col in self.within:
+                key.append(self.data[col].iloc[index])
+            key = tuple(key)
+            if key in cell_count:
+                cell_count[key] = cell_count[key] + 1
+            else:
+                cell_count[key] = 1
+        error_message = "Data is unbalanced."
+        if len(cell_count) != factor_levels:
+            raise ValueError(error_message)
+        count = cell_count[key]
+        for key in cell_count:
+            if count != cell_count[key]:
+                raise ValueError(error_message)
+        if self.data.shape[0] > count * factor_levels:
+            raise ValueError('There are more than 1 element in a cell! Missing'
+                             ' factors?')
+
+
+    def fit(self):
+        y = self.data[self.dv].values
+
+        def _not_slice(slices, slices_to_exclude, n):
+            ind = np.array([True]*n)
+            for term in slices_to_exclude:
+                s = slices[term]
+                ind[s] = False
+            return ind
+
+        def _ssr_reduced_model(y, x, term_slices, params, keys):
```

Comment:
I'm not a fan of nested functions except for trivial or necessary ones. This looks like it could be useful more generally and could go into the module (as private function).